packages/core/src/runtime/abstractRuntime.ts

```diff
@@ -0,0 +1,53 @@
+import { EventBus } from "../messaging/eventBus.ts";
+import type { Logger } from "../logging/logger.ts";
+import { RuntimeLogger } from "./RuntimeLogger.ts";
+
+export type SessionAccessorFunction = () => unknown;
+
+export interface RuntimeOptions {
+    loggers?: Logger[];
+    services?: Record<string, unknown>;
+    sessionAccessor?: SessionAccessorFunction;
+}
+
+export abstract class AbstractRuntime<TRoute = unknown, TNavigationItem = unknown> {
+    protected _logger: RuntimeLogger;
```

Comment:
Alex B mentionned it in later files, but we should add readonly in classes when its appropriate for protected fields! 

Also, it's really not that big of a deal, but you should use ECMAScript private fields

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Classes/Private_class_fields

The point being private only prevents access during compile time, but in most if not all cases, you don't want this field accessed during runtime either

Reply:
I moved everything to private fields using `#` except `AbstractRuntime`. It's a bummer to use 2 distinct syntaxes thought depending of the usage.